src/NServiceBus.Testing/Sagas/SagaMapper.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace NServiceBus.Testing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using NServiceBus.Sagas;
+
+    class SagaMapper : IConfigureHowToFindSagaWithMessage, IConfigureHowToFindSagaWithMessageHeaders
+    {
+        static Dictionary<Type, SagaMapper> sagaMappers = new Dictionary<Type, SagaMapper>();
```

Comment:
Have we made sure this is thread safe? XUnit for example parallelizes tests I think almost by default. NUnit allows to opt in for parallelism. Then we will run into concurrent modifications of the mappers

Reply:
Argh. I'll make all the Things into ConcurrentThings tomorrow.